frame/conviction-voting/src/types.rs

```diff
@@ -0,0 +1,235 @@
+		Self {
+			ayes: if vote.aye { votes } else { Zero::zero() },
+			nays: if vote.aye { Zero::zero() } else { votes },
+			turnout: capital,
+			dummy: PhantomData,
+		}
+	}
+
+	pub fn from_parts(ayes: Votes, nays: Votes, turnout: Votes) -> Self {
+		Self { ayes, nays, turnout, dummy: PhantomData }
+	}
+
+	/// Add an account's vote into the tally.
+	pub fn add(&mut self, vote: AccountVote<Votes>) -> Option<()> {
+		match vote {
+			AccountVote::Standard { vote, balance } => {
+				let Delegations { votes, capital } = vote.conviction.votes(balance);
+				self.turnout = self.turnout.checked_add(&capital)?;
+				match vote.aye {
+					true => self.ayes = self.ayes.checked_add(&votes)?,
+					false => self.nays = self.nays.checked_add(&votes)?,
+				}
+			},
+			AccountVote::Split { aye, nay } => {
+				let aye = Conviction::None.votes(aye);
+				let nay = Conviction::None.votes(nay);
+				self.turnout = self.turnout.checked_add(&aye.capital)?.checked_add(&nay.capital)?;
+				self.ayes = self.ayes.checked_add(&aye.votes)?;
+				self.nays = self.nays.checked_add(&nay.votes)?;
+			},
+		}
+		Some(())
+	}
+
+	/// Remove an account's vote from the tally.
+	pub fn remove(&mut self, vote: AccountVote<Votes>) -> Option<()> {
+		match vote {
+			AccountVote::Standard { vote, balance } => {
+				let Delegations { votes, capital } = vote.conviction.votes(balance);
+				self.turnout = self.turnout.checked_sub(&capital)?;
+				match vote.aye {
+					true => self.ayes = self.ayes.checked_sub(&votes)?,
+					false => self.nays = self.nays.checked_sub(&votes)?,
+				}
+			},
+			AccountVote::Split { aye, nay } => {
+				let aye = Conviction::None.votes(aye);
+				let nay = Conviction::None.votes(nay);
+				self.turnout = self.turnout.checked_sub(&aye.capital)?.checked_sub(&nay.capital)?;
+				self.ayes = self.ayes.checked_sub(&aye.votes)?;
+				self.nays = self.nays.checked_sub(&nay.votes)?;
+			},
+		}
+		Some(())
+	}
+
+	/// Increment some amount of votes.
+	pub fn increase(&mut self, approve: bool, delegations: Delegations<Votes>) -> Option<()> {
+		self.turnout = self.turnout.saturating_add(delegations.capital);
+		match approve {
+			true => self.ayes = self.ayes.saturating_add(delegations.votes),
+			false => self.nays = self.nays.saturating_add(delegations.votes),
+		}
+		Some(())
+	}
+
+	/// Decrement some amount of votes.
+	pub fn reduce(&mut self, approve: bool, delegations: Delegations<Votes>) -> Option<()> {
+		self.turnout = self.turnout.saturating_sub(delegations.capital);
+		match approve {
+			true => self.ayes = self.ayes.saturating_sub(delegations.votes),
+			false => self.nays = self.nays.saturating_sub(delegations.votes),
+		}
+		Some(())
+	}
+}
+
+/// Amount of votes and capital placed in delegation for an account.
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct Delegations<Balance> {
+	/// The number of votes (this is post-conviction).
+	pub votes: Balance,
```

Comment:
making a note here because, in general, it would make sense to me for this to be conviction, and then just calculate votes here, versus potentially someone messes up defining a `Delegations`, and messes up conviction voting.